# `hugoalh/no-alert`

Forbid use of [`alert`][ecmascript-alert].

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  alert();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis.alert();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis["alert"]();
  ```
- ```ts
  /* âŒ INVALID */
  window.alert();
  ```
- ```ts
  /* âŒ INVALID */
  window["alert"]();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis.window.alert();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis.window["alert"]();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis["window"].alert();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis["window"]["alert"]();
  ```
- ```ts
  /* âœ”ï¸ VALID */
  foo.alert();
  ```

## ğŸ“œ History

- *History before v0.4.0 is not displayed.*

## ğŸ“š Resources

- [ESLint rule `no-alert`](https://eslint.org/docs/latest/rules/no-alert)

[ecmascript-alert]: https://developer.mozilla.org/en-US/docs/Web/API/Window/alert
