# `hugoalh/no-confirm`

Forbid use of [`confirm`][ecmascript-confirm].

## üîß Options

*This rule does not have any option.*

## ‚ùå Invalid

- ```ts
  confirm();
  ```
- ```ts
  globalThis.confirm();
  ```
- ```ts
  globalThis["confirm"]();
  ```
- ```ts
  window.confirm();
  ```
- ```ts
  window["confirm"]();
  ```
- ```ts
  globalThis.window.confirm();
  ```
- ```ts
  globalThis.window["confirm"]();
  ```
- ```ts
  globalThis["window"].confirm();
  ```
- ```ts
  globalThis["window"]["confirm"]();
  ```

## ‚úîÔ∏è Valid

- ```ts
  foo.confirm();
  ```

## üìö Resources

- [ESLint rule `no-alert`](https://eslint.org/docs/latest/rules/no-alert)

[ecmascript-confirm]: https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm
