# `hugoalh/no-character-invisible`

Forbid character which is invisible.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const foo = "Hello,â€©world!";
                   //^ Unicode 002029
  /* âœ”ï¸ VALID */
  const foo = "Hello, world!";
                   //^ Unicode 000020
  ```

## ğŸ“œ History

- **v0.8.0:** Add.
