# `hugoalh/no-confirm`

Forbid use of [`confirm`][ecmascript-confirm].

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  confirm();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis.confirm();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis["confirm"]();
  ```
- ```ts
  /* âŒ INVALID */
  window.confirm();
  ```
- ```ts
  /* âŒ INVALID */
  window["confirm"]();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis.window.confirm();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis.window["confirm"]();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis["window"].confirm();
  ```
- ```ts
  /* âŒ INVALID */
  globalThis["window"]["confirm"]();
  ```
- ```ts
  /* âœ”ï¸ VALID */
  foo.confirm();
  ```

## ğŸ“œ History

- *History before v0.4.0 is not displayed.*

## ğŸ“š Resources

- [ESLint rule `no-alert`](https://eslint.org/docs/latest/rules/no-alert)

[ecmascript-confirm]: https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm
