# `hugoalh/no-character-ambiguous`

> ğŸ©¹ Fixer is available.

Forbid character which is ambiguous.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const foo = "ğŸ®ğŸ¯";

  /* âœ”ï¸ VALID */
  const foo = "23";
  ```

## ğŸ“œ History

- **v0.8.0:** Add.
