# `hugoalh/no-character-ambiguous`

> 🩹 Fixer is available.

Forbid character which is ambiguous.

## 🔧 Options

*This rule does not have any option.*

## ✍️ Examples

- ```ts
  /* ❌ INVALID */
  const foo = "𝟮𝟯";

  /* ✔️ VALID */
  const foo = "23";
  ```

## 📜 History

- **v0.8.0:** Add.
