# `hugoalh/no-duplicate-set-types`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid duplicate intersection and/or union types.

Intersection or union multiple same types have the same effect as single same type, which is a bad practice, cause confusion, and cause TypeScript Language Server consume more resources.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  type Foo = boolean & boolean & number & number & string & string & null & null;
  ```
- ```ts
  type Foo = boolean | boolean | number | number | string | string | null | null;
  ```
- ```ts
  type Foo = boolean | boolean & boolean | number | number | string | string | null | null;
  ```

## âœ”ï¸ Valid

- ```ts
  type Foo = boolean | number | string | null;
  ```
