# `hugoalh/no-duplicate-imports`

> âœ”ï¸ Default and recommended.

Forbid duplicate import sources.

Multiple imports with same source is a bad practice and cause confusion, which have the same effect as single import with same source, possibly not intended and is mergeable.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  import { a } from "./abc.ts";
  import { b } from "./abc.ts";
  import { c } from "./abc.ts";

  /* âœ”ï¸ VALID */
  import { a, b, c } from "./abc.ts";
  ```
- ```ts
  /* âœ”ï¸ VALID */
  import { a, b, c } from "./abc.ts";
  import { a, b, c } from "./abc.ts?debug";
  ```
- ```ts
  /* âœ”ï¸ VALID */
  import { a, b, c } from "./abc.ts";
  import { a, b, c } from "./abc.ts" with { key: "data" };
  ```
