# `hugoalh/no-duplicate-voids`

> ‚úîÔ∏è Default and recommended.

> ü©π Automatically fixable.

Forbid duplicate [`void`][ecmascript-void] operators.

Multiple [`void`][ecmascript-void] operators have the same effect as single [`void`][ecmascript-void] operator, possibly not intended.

## üîß Options

*This rule does not have any option.*

## ‚úçÔ∏è Examples

- ```ts
  /* ‚ùå INVALID */
  void void doSomething();

  /* ‚úîÔ∏è VALID */
  void doSomething();
  ```
- ```ts
  /* ‚ùå INVALID */
  void void void void void void void void void void doSomething();

  /* ‚úîÔ∏è VALID */
  void doSomething();
  ```
- ```ts
  /* ‚úîÔ∏è VALID */
  void (void doSomething()).doAnotherSomething();
  ```

[ecmascript-void]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void
