# `hugoalh/no-duplicate-awaits`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid duplicate [`await`][ecmascript-await] operators.

Multiple [`await`][ecmascript-await] operators have the same effect as single [`await`][ecmascript-await] operator, possibly not intended.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  await await doSomething();
  ```
- ```ts
  await await await await await await await await await await doSomething();
  ```

## âœ”ï¸ Valid

- ```ts
  await doSomething();
  ```

[ecmascript-await]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await
