# `hugoalh/no-duplicate-interface-contexts`

> âœ”ï¸ In the recommended rule set.

Forbid duplicate [`interface`][typescript-interface] contexts.

Multiple [`interface`][typescript-interface]s with same context is a bad practice and cause confusion, possibly mergeable.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  interface Foo {
    a: string;
    b: number;
  }
  interface Bar {
    a: string;
    b: number;
  }

  /* âœ”ï¸ VALID */
  interface Foo {
    a: string;
    b: number;
  }

  /* âœ”ï¸ VALID */
  interface Foo {
    a: string;
    b: number;
  }
  interface Bar<T extends string> {
    a: T;
    b: number;
  }
  ```

## ğŸ“œ History

- **v0.9.0:** Add.

[typescript-interface]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#interfaces
