# `hugoalh/no-duplicate-interfaces`

> âœ”ï¸ Default and recommended.

Forbid duplicate [`interface`][typescript-interface]s.

Multiple [`interface`][typescript-interface]s with same identifier is a bad practice and cause confusion, which have the same effect as single [`interface`][typescript-interface] with same identifier.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  interface Foo {
    a: string;
    b: string;
  }
  ...
  ...
  ...
  interface Foo {
    c: string;
    d: string;
  }
  ```

## âœ”ï¸ Valid

- ```ts
  interface Foo {
    a: string;
    b: string;
    c: string;
    d: string;
  }
  ```
  
[typescript-interface]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#interfaces
