# `hugoalh/no-duplicate-voids`

> âœ”ï¸ In the recommended rule set.

> ğŸ©¹ Fixer is available.

Forbid duplicate [`void`][ecmascript-void]s.

Multiple [`void`][ecmascript-void]s have the same effect as single [`void`][ecmascript-void], possibly not intended.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  void void doSomething();

  /* âœ”ï¸ VALID */
  void doSomething();
  ```
- ```ts
  /* âŒ INVALID */
  void void void void void void void void void void doSomething();

  /* âœ”ï¸ VALID */
  void doSomething();
  ```

## ğŸ“œ History

- **v0.7.0:** Add.

[ecmascript-void]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void
