# `hugoalh/no-duplicate-typeofs`

> âœ”ï¸ Default and recommended.

> ðŸ©¹ Automatically fixable.

Forbid duplicate [`typeof`][es-typeof] operators.

Multiple [`typeof`][es-typeof] operators always return `"string"`, possibly not intended.

## ðŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  typeof typeof globalThis;
  //=> "string"
  ```
- ```ts
  typeof typeof typeof typeof typeof typeof typeof typeof typeof typeof globalThis;
  //=> "string"
  ```

## âœ”ï¸ Valid

- ```ts
  typeof globalThis;
  //=> "object"
  ```

[es-typeof]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof
