# `hugoalh/no-duplicate-awaits`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid duplicate [`await`][ecmascript-await] operators.

Multiple [`await`][ecmascript-await] operators have the same effect as single [`await`][ecmascript-await] operator, possibly not intended.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  await await doSomething();

  /* âœ”ï¸ VALID */
  await doSomething();
  ```
- ```ts
  /* âŒ INVALID */
  await await await await await await await await await await doSomething();

  /* âœ”ï¸ VALID */
  await doSomething();
  ```
- ```ts
  /* âœ”ï¸ VALID */
  await (await doSomething()).doAnotherSomething();
  ```

[ecmascript-await]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await
