# `hugoalh/no-duplicate-typeofs`

> âœ”ï¸ Default and recommended.

> ðŸ©¹ Automatically fixable.

Forbid duplicate [`typeof`][ecmascript-typeof] operators.

Multiple [`typeof`][ecmascript-typeof] operators always return `"string"`, possibly not intended.

## ðŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  typeof typeof globalThis;
  //=> "string"

  /* âœ”ï¸ VALID */
  typeof globalThis;
  //=> "object"
  ```
- ```ts
  /* âŒ INVALID */
  typeof typeof typeof typeof typeof typeof typeof typeof typeof typeof globalThis;
  //=> "string"

  /* âœ”ï¸ VALID */
  typeof globalThis;
  //=> "object"
  ```

## ðŸ“œ History

- **v0.5.0:** Add.

[ecmascript-typeof]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof
