# `hugoalh/no-duplicate-types`

> âœ”ï¸ Default and recommended.

Forbid duplicate [`type`][typescript-type]s.

Multiple [`type`][typescript-type]s with same context is a bad practice and cause confusion, possibly not intended.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  type A = boolean | string;
  ...
  ...
  ...
  type B = boolean | string;
  ```

## âœ”ï¸ Valid

- ```ts
  type A = boolean | string;
  ```

[typescript-type]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-aliases
