# `hugoalh/no-enum`

> âœ”ï¸ Default and recommended.

Forbid use of [`enum`][typescript-enum].

[`enum` is not type safe.](https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh)

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  enum Foo {
    ONE = "one",
    TWO = "two"
  }

  /* âœ”ï¸ VALID */
  type Foo = "one" | "two";
  ```
- ```ts
  /* âŒ INVALID */
  enum Roles {
    Admin,
    Writer,
    Reader
  }

  /* âœ”ï¸ VALID */
  type Roles = "admin" | "writer" | "reader";
  ```

[typescript-enum]: https://www.typescriptlang.org/docs/handbook/enums.html
