# `hugoalh/no-enum`

> âœ”ï¸ Default and recommended.

Forbid use of [`enum`](https://www.typescriptlang.org/docs/handbook/enums.html).

[`enum` is not type safe.](https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh)

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  enum Foo {
    ONE = "one",
    TWO = "two"
  }
  ```
- ```ts
  enum Roles {
    Admin,
    Writer,
    Reader
  }
  ```

## âœ”ï¸ Valid

- ```ts
  type Foo = "one" | "two";
  ```
- ```ts
  type Roles = "admin" | "writer" | "reader";
  ```
