# `hugoalh/no-empty-yield`

> âœ”ï¸ Default and recommended.

Forbid empty [`yield`][es-yield].

Empty [`yield`][es-yield] possibly missing the expression.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  function* foo() {
    doSomething();
    yield;
  }
  ```
- ```ts
  function* foo() {
    doSomething();
    yield;
    yield;
    yield;
    yield;
    yield;
  }
  ```

## âœ”ï¸ Valid

- ```ts
  function* foo() {
    doSomething();
    yield undefined;
  }
  ```

[es-yield]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield
