# `hugoalh/no-empty-yield`

> ‚úîÔ∏è Default and recommended.

Forbid empty [`yield`][ecmascript-yield].

Empty [`yield`][ecmascript-yield] possibly missing the expression.

## üîß Options

*This rule does not have any option.*

## ‚ùå Invalid

- ```ts
  function* foo() {
    doSomething();
    yield;
  }
  ```
- ```ts
  function* foo() {
    doSomething();
    yield;
    yield;
    yield;
    yield;
    yield;
  }
  ```

## ‚úîÔ∏è Valid

- ```ts
  function* foo() {
    doSomething();
    yield undefined;
  }
  ```

[ecmascript-yield]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield
