# `hugoalh/no-empty-yield`

> âœ”ï¸ In the recommended rule set.

Forbid empty [`yield`][ecmascript-yield].

Empty [`yield`][ecmascript-yield] possibly missing the expression.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  function* foo() {
    doSomething();
    yield;
  }
  ```
- ```ts
  /* âŒ INVALID */
  function* foo() {
    doSomething();
    yield;
    yield;
    yield;
    yield;
    yield;
  }
  ```
- ```ts
  /* âœ”ï¸ VALID */
  function* foo() {
    doSomething();
    yield undefined;
  }
  ```

## ğŸ“œ History

- **v0.5.0:** Add.

[ecmascript-yield]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield
