# `hugoalh/no-irregular-numeric-exponent-case`

> âœ”ï¸ In the recommended rule set.

> ðŸ©¹ Fixer is available.

Forbid irregular numeric exponent case.

## ðŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const foo = 1E4;

  /* âœ”ï¸ VALID */
  const foo = 1e4;
  ```
- ```ts
  /* âŒ INVALID */
  const foo = 12.3E4;

  /* âœ”ï¸ VALID */
  const foo = 12.3e4;
  ```
- ```ts
  /* âœ”ï¸ VALID */
  const foo = 0x123E4;
  //=> 74724
  ```

## ðŸ“œ History

- **v0.9.0:** Add.
