# `hugoalh/no-import-https`

Forbid import module via protocol `https:`.

> [!NOTE]
> - Due to the Deno linter framework limitations, detect imports from imports map is not possible.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  import x from "https://example.com/x.ts";

  /* âœ”ï¸ VALID */
  import x from "./x.ts";
  ```

## ğŸ“œ History

- *History before v0.4.0 is not displayed.*
