# `hugoalh/no-import-node`

Forbid import module via protocol `node:`.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  import path from "node:path";
  ```

## âœ”ï¸ Valid

- ```ts
  import path from "jsr:@std/path@^1.0.8";
  ```
