# `hugoalh/no-import-jsr`

> ‚úîÔ∏è Default and recommended.

Forbid import JSR module.

Default to only forbid import JSR module via URL.

## üîß Options

### `viaProtocol`

`{boolean = false}` Whether to forbid import JSR module via protocol `jsr:`.

### `viaURL`

`{boolean = true}` Whether to forbid import JSR module via URL.

## ‚úçÔ∏è Examples

- ```ts
  /* ‚ùå INVALID */
  import { copy } from "https://jsr.io/@std/fs/1.0.14/copy.ts";

  /* ‚úîÔ∏è VALID */
  import { copy } from "jsr:@std/fs@^1.0.14/copy";
  ```
- ```ts
  ---
  viaProtocol: true
  ---
  /* ‚ùå INVALID */
  import { copy } from "jsr:@std/fs@^1.0.14/copy";
  ```
