# `hugoalh/no-import-self`

> âœ”ï¸ In the recommended rule set.

Forbid the module import itself.

> [!NOTE]
> - Due to the Deno linter plugin API limitations, detect imports from imports map is not possible.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */

  /* foo.ts */
  import foo from "./foo.ts";



  /* âœ”ï¸ VALID */

  /* foo.ts */
  import bar from "./bar.ts";
  ```

## ğŸ“œ History

- **v0.4.0:** Add.

## ğŸ“š References

- [ESLint rule `import/no-self-import`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-self-import.md)
