# `hugoalh/no-import-file`

> âœ”ï¸ In the recommended rule set.

> ğŸ©¹ Fixer is available.

Forbid import module via protocol `file:`.

Import module via protocol `file:` is a bad practice and unnecessary as it ties the code using it to your computer, thus makes it unusable in packages distribution for instance.

> [!NOTE]
> - Due to the Deno linter framework limitations, detect imports from imports map is not possible.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  import x from "file:///C:/path/to/the/file.ts";

  /* âœ”ï¸ VALID */
  import x from "../path/to/the/file.ts";
  ```

## ğŸ“œ History

- *History before v0.4.0 is not displayed.*
