# `hugoalh/no-import-self`

> âœ”ï¸ Default and recommended.

Forbid the module import itself.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  /* foo.ts */
  import foo from "./foo.ts";
  ```

## âœ”ï¸ Valid

- ```ts
  /* foo.ts */
  import bar from "./bar.ts";
  ```

## ğŸ“š References

- [ESLint rule `import/no-self-import`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-self-import.md)
