# `hugoalh/no-import-http`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid import module via protocol `http:`.

Import module via protocol `http:` is not secure.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  import x from "http://example.com/x.ts";
  ```

## âœ”ï¸ Valid

- ```ts
  import x from "https://example.com/x.ts";
  ```
