# `hugoalh/no-irregular-numeric-separation`

> âœ”ï¸ In the recommended rule set.

> ğŸ©¹ Fixer is available.

Forbid irregular numeric separation.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const foo = 12_34567_890n;

  /* âœ”ï¸ VALID */
  const foo = 1_234_567_890n;

  /* âœ”ï¸ VALID */
  const foo = 1234567890n;
  ```
- ```ts
  /* âŒ INVALID */
  const foo = 12_34567_890;

  /* âœ”ï¸ VALID */
  const foo = 1_234_567_890;

  /* âœ”ï¸ VALID */
  const foo = 1234567890;
  ```

## ğŸ“œ History

- **v0.9.0:** Add.
