# `hugoalh/no-iife`

Forbid use of [immediately invoked function expression (IIFE)][ecmascript-iife].

## ðŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  (function () {
    // Statements...
  })();
  ```
- ```ts
  /* âŒ INVALID */
  (async function () {
    // Statements...
  })();
  ```
- ```ts
  /* âŒ INVALID */
  (() => {
    // Statements...
  })();
  ```
- ```ts
  /* âŒ INVALID */
  (async () => {
    // Statements...
  })();
  ```

[ecmascript-iife]: https://developer.mozilla.org/en-US/docs/Glossary/IIFE
