# `hugoalh/no-import-npm`

> ü©π Automatically fixable.

Forbid import NPM module.

## üîß Options

### `viaProtocol`

`{boolean = true}` Whether to forbid import NPM module via protocol `npm:`.

### `viaURL`

`{boolean = true}` Whether to forbid import NPM module via URL.

## ‚úçÔ∏è Examples

- ```ts
  /* ‚ùå INVALID */
  import confetti from "npm:canvas-confetti@^1.6.0";

  /* ‚úîÔ∏è VALID */
  import confetti from "./confetti.ts";
  ```
- ```ts
  /* ‚ùå INVALID */
  import confetti from "https://esm.sh/canvas-confetti@^1.6.0";

  /* ‚úîÔ∏è VALID */
  import confetti from "./confetti.ts";
  ```
