# `hugoalh/no-import-http`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid import module via protocol `http:`.

Import module via protocol `http:` is not secure.

> [!NOTE]
> - Due to the Deno linter framework limitations, detect imports from imports map is not possible.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  import x from "http://example.com/x.ts";

  /* âœ”ï¸ VALID */
  import x from "https://example.com/x.ts";
  ```
