# `hugoalh/no-import-absolute`

> âœ”ï¸ In the recommended rule set.

> ğŸ©¹ Fixer is available.

Forbid import module via absolute path.

Import module via absolute path is a bad practice as it ties the code using it to your computer, thus makes it unusable in packages distribution for instance.

> [!NOTE]
> - Due to the Deno linter framework limitations, detect imports from imports map is not possible.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  import f from "/foo.ts";

  /* âœ”ï¸ VALID */
  import f from "./foo.ts";
  ```
- ```ts
  /* âŒ INVALID */
  import f from "/some/path.ts";

  /* âœ”ï¸ VALID */
  import f from "../some/path.ts";
  ```

## ğŸ“œ History

- **v0.4.0:** Add.

## ğŸ“š References

- [ESLint rule `import/no-absolute-path`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-absolute-path.md)
