# `hugoalh/no-import-node`

Forbid import module via protocol `node:`.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  import path from "node:path";

  /* âœ”ï¸ VALID */
  import path from "jsr:@std/path@^1.0.8";
  ```
