# `hugoalh/no-import-https`

Forbid import module via protocol `https:`.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  import x from "https://example.com/x.ts";
  ```

## âœ”ï¸ Valid

- ```ts
  import x from "./x.ts";
  ```
