# `hugoalh/no-import-file`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid import module via protocol `file:`.

Import module via protocol `file:` is a bad practice and unnecessary as it ties the code using it to your computer, thus makes it unusable in packages distribution for instance.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  import x from "file:///C:/path/to/the/file.ts";

  /* âœ”ï¸ VALID */
  import x from "../path/to/the/file.ts";
  ```
