# `hugoalh/no-import-data`

> âœ”ï¸ Default and recommended.

Forbid import module via protocol `data:`.

Import module via protocol `data:` is a bad practice as it is hard to maintenance and not secure.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */

  /* mod.ts */
  import x from "data:text/javascript,export default 42;";



  /* âœ”ï¸ VALID */

  /* mod.ts */
  const x = 42;
  ```
- ```ts
  /* âœ”ï¸ VALID */

  /* x.ts */
  export default 42;

  /* mod.ts */
  import x from "./x.ts";
  ```
