# `hugoalh/no-import-absolute`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid import module via absolute path.

Import module via absolute path is a bad practice as it ties the code using it to your computer, and therefore makes it unusable in packages distribution for instance.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  import f from "/foo.ts";
  ```
- ```ts
  import f from "/some/path.ts";
  ```

## âœ”ï¸ Valid

- ```ts
  import f from "./foo.ts";
  ```
- ```ts
  import f from "../some/path.ts";
  ```

## ğŸ“š References

- [ESLint rule `import/no-absolute-path`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-absolute-path.md)
