# `hugoalh/no-import-https`

Forbid import module via protocol `https:`.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  import x from "https://example.com/x.ts";

  /* âœ”ï¸ VALID */
  import x from "./x.ts";
  ```
