# `hugoalh/no-import-dynamic`

Forbid import module dynamically.

This is designed for the projects which use bundler, as import modules dynamically maybe cause bundler unable to bundle all of the needed modules.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const x = await import("https://example.com/x.ts");

  /* âœ”ï¸ VALID */
  import x from "https://example.com/x.ts";
  ```

## ğŸ“œ History

- **v0.5.0:** Add.
