# `hugoalh/no-modifier-public`

> âœ”ï¸ In the recommended rule set.

Forbid modifier [`public`][typescript-public].

Use of modifier [`public`][typescript-public] is useless as public is the default visibility.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  class Foo {
    public value: string;
    public constructor() {
      this.value = "bar";
    }
  }

  /* âœ”ï¸ VALID */
  class Foo {
    value: string;
    constructor() {
      this.value = "bar";
    }
  }
  ```

## ğŸ“œ History

- **v0.8.0:** Add.

[typescript-public]: https://www.typescriptlang.org/docs/handbook/2/classes.html#public
