# `hugoalh/no-ternary-nest`

> [!CAUTION]
> - This rule is replaced by rule [`hugoalh/max-nest-ternary`][rule-hugoalh-max-nest-ternary] since v0.9.0.

Forbid nested [ternary expression][ecmascript-ternary].

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const thing = foo ? bar : ((baz === qux) ? quxx : foobar);
  ```
- ```ts
  /* âŒ INVALID */
  foo ? ((baz === qux) ? quxx() : foobar()) : bar();
  ```
- ```ts
  /* âœ”ï¸ VALID */
  const thing = foo ? bar : foobar;
  ```

## ğŸ“œ History

- **v0.9.0:** Replace by rule [`hugoalh/max-nest-ternary`][rule-hugoalh-max-nest-ternary].
- *History before v0.4.0 is not displayed.*

## ğŸ“š Resources

- [ESLint rule `no-nested-ternary`](https://eslint.org/docs/latest/rules/no-nested-ternary)

[ecmascript-ternary]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator
[rule-hugoalh-max-nest-ternary]: https://github.com/hugoalh/deno-lint-rules/blob/main/docs/rules/max-nest-ternary.md
