# `hugoalh/no-useless-class-constructor`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid useless [class constructor][ecmascript-class-constructor].

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  class Foo {
    constructor() {
    }
  }
  ```

## âœ”ï¸ Valid

- ```ts
  class Foo {
    constructor() {
		doSomething();
	}
  }
  ```

## ğŸ“š References

- [ESLint rule `no-useless-constructor`](https://eslint.org/docs/latest/rules/no-useless-constructor)

[ecmascript-class-constructor]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor
