# `hugoalh/no-useless-block`

> âœ”ï¸ In the recommended rule set.

> ðŸ©¹ Fixer is available.

Forbid useless block.

## ðŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  function foo(a) {
    {
      doSomething(a);
    }
  }

  /* âœ”ï¸ VALID */
  function foo(a) {
    doSomething(a);
  }
  ```
- ```ts
  /* âŒ INVALID */
  function foo(a) {
    doSomething(a);
    {
      doAnotherSomething(a);
    }
  }

  /* âœ”ï¸ VALID */
  function foo(a) {
    doSomething(a);
    doAnotherSomething(a);
  }
  ```
- ```ts
  /* âœ”ï¸ VALID */
  const c = 1;
  {
    const c = 2;
  }
  console.log(c);
  //=> 1
  ```

## ðŸ“œ History

- **v0.6.0:** Also detect [`switch`][ecmascript-switch] case with empty block.
- **v0.4.0:** Add.

[ecmascript-switch]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch
