# `hugoalh/no-unsafe-number`

> âœ”ï¸ Default and recommended.

Forbid unsafe number.

Number literals with values equal to 2<sup>53</sup> or greater are too large to be represented accurately.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  const foo = 98765432109876543210;
  ```
- ```ts
  const bar = 98765432109876543.2109;
  ```

## âœ”ï¸ Valid

- ```ts
  const foo = 98765432109876543210n;
  ```
- ```ts
  const bar = 9876543.2109;
  ```
