# `hugoalh/no-useless-typealias`

> [!CAUTION]
> - This rule is renamed to [`hugoalh/no-useless-type`][rule-no-useless-type] since v0.7.0.

> âœ”ï¸ In the recommended rule set.

Forbid useless [type alias][typescript-typealias].

Declare the [type alias][typescript-typealias] with simple type is a bad practice, cause confusion, cause TypeScript Language Server consume more resources, pointless, and unnecessary.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  type Foo = any;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = bigint;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = boolean;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = never;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = null;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = number;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = object;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = string;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = symbol;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = undefined;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = unknown;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = void;
  ```
- ```ts
  /* âŒ INVALID */
  type Foo = Body;
  ```
- ```ts
  /* âœ”ï¸ VALID */
  type Foo = number | string;
  ```

## ğŸ“œ History

- **v0.7.0:** Rename to [`hugoalh/no-useless-type`][rule-no-useless-type].
- **v0.5.0:** Add.

[rule-no-useless-type]: https://github.com/hugoalh/deno-lint-rules/blob/main/docs/rules/no-useless-type.md
[typescript-typealias]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-aliases
