# `hugoalh/no-useless-typealias`

> [!CAUTION]
> - This rule is renamed to rule [`hugoalh/no-useless-type`][rule-hugoalh-no-useless-type] since v0.7.0.

> ‚úîÔ∏è In the recommended rule set.

Forbid useless [type alias][typescript-typealias].

Declare the [type alias][typescript-typealias] with simple type is a bad practice, cause confusion, cause TypeScript Language Server consume more resources, pointless, and unnecessary.

## üîß Options

*This rule does not have any option.*

## ‚úçÔ∏è Examples

- ```ts
  /* ‚ùå INVALID */
  type Foo = any;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = bigint;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = boolean;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = never;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = null;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = number;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = object;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = string;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = symbol;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = undefined;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = unknown;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = void;
  ```
- ```ts
  /* ‚ùå INVALID */
  type Foo = Body;
  ```
- ```ts
  /* ‚úîÔ∏è VALID */
  type Foo = number | string;
  ```

## üìú History

- **v0.7.0:** Rename to rule [`hugoalh/no-useless-type`][rule-hugoalh-no-useless-type].
- **v0.5.0:** Add.

[rule-hugoalh-no-useless-type]: https://github.com/hugoalh/deno-lint-rules/blob/main/docs/rules/no-useless-type.md
[typescript-typealias]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-aliases
