# `hugoalh/no-use-strict`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid use of [`use strict`][ecmascript-strict] directive as ECMAScript modules always have strict mode semantics.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  "use strict";

  // strict mode

  function foo() {
    // strict mode
  }
  ```
- ```ts
  function foo() {
    "use strict";
    // strict mode
  }
  ```
- ```ts
  (function() {
    "use strict";
    function bar() {
      // strict mode
    }
  })();
  ```

## âœ”ï¸ Valid

- ```ts
  const foo = "use strict";
  ```
- ```ts
  console.log("use strict");
  ```

## ğŸ“š References

- [ESLint rule `strict`](https://eslint.org/docs/latest/rules/strict)

[ecmascript-strict]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode
