# `hugoalh/no-useless-class-constructor`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid useless [class constructor][ecmascript-class-constructor].

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  class Foo {
    constructor() {
    }
    ...
  }

  /* âœ”ï¸ VALID */
  class Foo {
    constructor() {
      doSomething();
    }
    ...
  }

  /* âœ”ï¸ VALID */
  class Foo {
    ...
  }
  ```
- ***(>= v0.7.0)***
  ```ts
  /* âŒ INVALID */
  class Foo extends Bar {
    constructor() {
      super();
    }
    ...
  }

  /* âœ”ï¸ VALID */
  class Foo extends Bar {
    constructor(a, b) {
      super(a, b, c);
    }
    ...
  }

  /* âœ”ï¸ VALID */
  class Foo extends Bar {
    ...
  }
  ```

## ğŸ“œ History

- **v0.4.0:** Add and rename from `hugoalh/no-empty-class-constructor`.

## ğŸ“š References

- [ESLint rule `no-useless-constructor`](https://eslint.org/docs/latest/rules/no-useless-constructor)

[ecmascript-class-constructor]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor
