# `hugoalh/no-useless-try`

> ‚úîÔ∏è Default and recommended.

> ü©π Automatically fixable.

Forbid useless [`try-catch-finally`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) statement.

## üîß Options

*This rule does not have any option.*

## ‚ùå Invalid

- ```ts
  try {
    doSomethingThatMightThrow();
  } catch (e) {
    throw e;
  }
  ```
- ```ts
  try {
    doSomethingThatMightThrow();
  } catch (e) {
    throw e;
  } finally {
    cleanUp();
  }
  ```

## ‚úîÔ∏è Valid

- ```ts
  try {
    doSomethingThatMightThrow();
  } catch (e) {
    doSomethingBeforeRethrow();
    throw e;
  }
  ```
- ```ts
  try {
    doSomethingThatMightThrow();
  } catch (e) {
    handleError(e);
  }
  ```
- ```ts
  try {
    doSomethingThatMightThrow();
  } finally {
    cleanUp();
  }
  ```

## üìö References

- [ESLint rule `no-useless-catch`](https://eslint.org/docs/latest/rules/no-useless-catch)
