# `hugoalh/no-use-strict`

> âœ”ï¸ In the recommended rule set.

> ğŸ©¹ Fixer is available.

Forbid use of [`use strict`][ecmascript-strict] directive as ECMAScript modules always have strict mode semantics.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  "use strict";

  // strict mode

  function foo() {
    // strict mode
  }
  ```
- ```ts
  /* âŒ INVALID */
  function foo() {
    "use strict";
    // strict mode
  }
  ```
- ```ts
  /* âŒ INVALID */
  (function() {
    "use strict";
    function bar() {
      // strict mode
    }
  })();
  ```
- ```ts
  /* âœ”ï¸ VALID */
  const foo = "use strict";
  ```
- ```ts
  /* âœ”ï¸ VALID */
  console.log("use strict");
  ```

## ğŸ“œ History

- **v0.4.0:** Add.

## ğŸ“š References

- [ESLint rule `strict`](https://eslint.org/docs/latest/rules/strict)

[ecmascript-strict]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode
