# `hugoalh/no-useless-block`

> âœ”ï¸ Default and recommended.

> ðŸ©¹ Automatically fixable.

Forbid useless block.

## ðŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  function foo(a) {
    {
      doSomething(a);
    }
  }
  ```

## âœ”ï¸ Valid

- ```ts
  function foo(a) {
    doSomething(a);
  }
  ```
- ```ts
  function foo(a) {
	doSomething(a);
    {
      doAnotherSomething(a);
    }
  }
  ```
- ```ts
  const c = 1;
  {
    const c = 2;
  }
  console.log(c);
  //=> 1
  ```
