# `hugoalh/no-useless-expression`

> âœ”ï¸ Default and recommended.

Forbid useless expression which will do nothing, likely missing the assignment or call.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  doSomething;
  ```
- ```ts
  "Hello, world!";
  ```

## âœ”ï¸ Valid

- ```ts
  doSomething();
  ```
- ```ts
  const foo = "Hello, world!";
  ```

## ğŸ“š References

- [ESLint rule `no-unused-expressions`](https://eslint.org/docs/latest/rules/no-unused-expressions)
