# `hugoalh/no-useless-ternary`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid useless [ternary expression][ecmascript-operator-ternary].

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  const isYes = (answer === 1) ? true : false;
  ```
- ```ts
  const isNo = (answer === 1) ? false : true;
  ```
- ```ts
  const a = (x === 2) ? true : false;
  ```
- ```ts
  const b = x ? true : false;
  ```
- ```ts
  const b = x ? 10 : 10;
  ```

## âœ”ï¸ Valid

- ```ts
  const isYes = answer === 1;
  ```
- ```ts
  const isNo = answer !== 1;
  ```
- ```ts
  const a = x === 2;
  ```
- ```ts
  const b = Boolean(x);
  ```
- ```ts
  const b = 10;
  ```

## ğŸ“š References

- [ESLint rule `no-unneeded-ternary`](https://eslint.org/docs/latest/rules/no-unneeded-ternary)

[ecmascript-operator-ternary]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator
