# `hugoalh/no-useless-expression`

> âœ”ï¸ Default and recommended.

Forbid useless expression which will do nothing, possibly missing the assignment or call.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  doSomething;

  /* âœ”ï¸ VALID */
  doSomething();
  ```
- ```ts
  /* âŒ INVALID */
  "Hello, world!";

  /* âœ”ï¸ VALID */
  const foo = "Hello, world!";
  ```

## ğŸ“š References

- [ESLint rule `no-unused-expressions`](https://eslint.org/docs/latest/rules/no-unused-expressions)
