# `hugoalh/no-useless-try`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid useless [`try`][ecmascript-try] statement.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  try {
    doSomethingThatMightThrow();
  } catch (e) {
    throw e;
  }

  /* âœ”ï¸ VALID */
  doSomethingThatMightThrow();
  ```
- ```ts
  /* âŒ INVALID */
  try {
    doSomethingThatMightThrow();
  } catch (e) {
    throw e;
  } finally {
    cleanUp();
  }

  /* âœ”ï¸ VALID */
  try {
    doSomethingThatMightThrow();
  } finally {
    cleanUp();
  }
  ```
- ```ts
  /* âœ”ï¸ VALID */
  try {
    doSomethingThatMightThrow();
  } catch (e) {
    doSomethingBeforeRethrow();
    throw e;
  }
  ```
- ```ts
  /* âœ”ï¸ VALID */
  try {
    doSomethingThatMightThrow();
  } catch (e) {
    handleError(e);
  }
  ```

## ğŸ“œ History

- **v0.4.0:** Add.

## ğŸ“š References

- [ESLint rule `no-useless-catch`](https://eslint.org/docs/latest/rules/no-useless-catch)

[ecmascript-try]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch
