# `hugoalh/no-useless-export`

> âœ”ï¸ In the recommended rule set.

> ğŸ©¹ Fixer is available.

Forbid useless [`export`][ecmascript-export] statement.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  export {};
  ```
- ```ts
  /* âŒ INVALID */
  import "some-other-module";
  export {};
  ```
- ```ts
  /* âœ”ï¸ VALID */
  export const value = "Hello, world!";
  ```

## ğŸ“œ History

- **v0.4.0:** Add.

## ğŸ“š References

- [ESLint rule `@typescript-eslint/no-useless-empty-export`](https://typescript-eslint.io/rules/no-useless-empty-export/)

[ecmascript-export]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export
