# `hugoalh/no-useless-export`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid useless [`export`][ecmascript-export] statement.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  export {};
  ```
- ```ts
  /* âŒ INVALID */
  import "some-other-module";
  export {};
  ```
- ```ts
  /* âœ”ï¸ VALID */
  export const value = "Hello, world!";
  ```

## ğŸ“š References

- [ESLint rule `@typescript-eslint/no-useless-empty-export`](https://typescript-eslint.io/rules/no-useless-empty-export/)

[ecmascript-export]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export
