# `hugoalh/no-useless-class-static-block`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Forbid useless [class static (initialization) block][ecmascript-class-static-block].

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  class Foo {
    static {
    }
  }
  ```

## âœ”ï¸ Valid

- ```ts
  class Foo {
    static {
      doSomething();
    }
  }
  ```

## ğŸ“š References

- [ESLint rule `no-empty-static-block`](https://eslint.org/docs/latest/rules/no-empty-static-block)

[ecmascript-class-static-block]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Static_initialization_blocks
