# `hugoalh/prefer-variable-declaration-ungroup-form`

> ğŸ©¹ Fixer is available.

Prefer declare variable in ungroup form.

Group variable declaration in the initialize of the statement [`for`][ecmascript-for] is always permit.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const a = 1, b = 2, c = 3;

  /* âœ”ï¸ VALID */
  const a = 1;
  const b = 2;
  const c = 3;
  ```

## ğŸ“œ History

- **v0.9.0:** Add.

[ecmascript-for]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for
