# `hugoalh/prefer-regexp-flag-unicode`

Prefer the [regular expression][ecmascript-regexp] is contain Unicode flag (`u` or `v`).

## ğŸ”§ Options

### `flag`

`{DenoLintRulePreferRegExpFlagUnicodeType = "u"}` Select which regular expression Unicode flag should use.

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const foo = /aaa/;

  /* âœ”ï¸ VALID */
  const foo = /aaa/u;
  ```
- ```ts
  /* âŒ INVALID */
  const foo = /bbb/gi;

  /* âœ”ï¸ VALID */
  const foo = /bbb/giu;
  ```

## ğŸ“œ History

- *History before v0.4.0 is not displayed.*

## ğŸ“š Resources

- [ESLint rule `require-unicode-regexp`](https://eslint.org/docs/latest/rules/require-unicode-regexp)

[ecmascript-regexp]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions
