# `hugoalh/prefer-ascii-identifier`

> âœ”ï¸ Default and recommended.

Prefer ASCII identifier, an alternative of the Deno lint rule [`prefer-ascii`](https://docs.deno.com/lint/rules/prefer-ascii/) which only enforce on the identifier.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  const Ï€ = Math.PI;
  ```
- ```ts
  function ã“ã‚“ã«ã¡ã¯(åå‰: string) {
    console.log(`ã“ã‚“ã«ã¡ã¯ã€${åå‰}ã•ã‚“`);
  }
  ```
- ```ts
  class CafÃ¨ {}
  ```

## âœ”ï¸ Valid

- ```ts
  const pi = Math.PI;
  ```
- ```ts
  function hello(name: string) {
    console.log(`ã“ã‚“ã«ã¡ã¯ã€${name}ã•ã‚“`);
  }
  ```
- ```ts
  class Cafe {}
  ```
