# `hugoalh/prefer-interface`

> âœ”ï¸ In the recommended rule set.

> ğŸ©¹ Fixer is available.

Prefer to use [`interface`][typescript-interface] instead of [`type`][typescript-typealias].

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  type T = { x: number };

  /* âœ”ï¸ VALID */
  interface T {
    x: number;
  }
  ```
- ```ts
  /* âœ”ï¸ VALID */
  type Foo = string | {};
  ```
- ```ts
  /* âœ”ï¸ VALID */
  type T = boolean | string;
  ```

## ğŸ“œ History

- **v0.4.0:** Add.

## ğŸ“š References

- [ESLint rule `@typescript-eslint/consistent-type-definitions`](https://typescript-eslint.io/rules/consistent-type-definitions/)
- Typescript
  - [Differences between type aliases and interfaces](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces)

[typescript-interface]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#interfaces
[typescript-typealias]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-aliases
