# `hugoalh/prefer-interface`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Prefer to use `interface` instead of `type`.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  type T = { x: number };
  ```

## âœ”ï¸ Valid

- ```ts
  interface T {
    x: number;
  }
  ```
- ```ts
  type Foo = string | {};
  ```
- ```ts
  type T = string;
  ```

## ğŸ“š References

- [ESLint rule `@typescript-eslint/consistent-type-definitions`](https://typescript-eslint.io/rules/consistent-type-definitions/)
