# `hugoalh/prefer-ascii-identifier`

> âœ”ï¸ Default and recommended.

Prefer ASCII identifier, an alternative of the Deno lint rule [`prefer-ascii`](https://docs.deno.com/lint/rules/prefer-ascii/) which only enforce on the identifier.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  const Ï€ = Math.PI;

  /* âœ”ï¸ VALID */
  const pi = Math.PI;
  ```
- ```ts
  /* âŒ INVALID */
  function ã“ã‚“ã«ã¡ã¯(åå‰: string) {
    console.log(`ã“ã‚“ã«ã¡ã¯ã€${åå‰}ã•ã‚“`);
  }

  /* âœ”ï¸ VALID */
  function hello(name: string) {
    console.log(`ã“ã‚“ã«ã¡ã¯ã€${name}ã•ã‚“`);
  }
  ```
- ```ts
  /* âŒ INVALID */
  class CafÃ¨ {}

  /* âœ”ï¸ VALID */
  class Cafe {}
  ```
