# `hugoalh/prefer-statement-block`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Prefer curly braces around statement blocks.

## ğŸ”§ Options

*This rule does not have any option.*

## âŒ Invalid

- ```ts
  if (foo) foo++;
  ```
- ```ts
  if (foo) {
    baz();
  } else qux();
  ```
- ```ts
  while (bar)
    baz();
  ```

## âœ”ï¸ Valid

- ```ts
  if (foo) {
    foo++;
  }
  ```
- ```ts
  if (foo) {
    baz();
  } else {
    qux();
  }
  ```
- ```ts
  while (bar) {
    baz();
  }
  ```

## ğŸ“š References

- [ESLint rule `curly`](https://eslint.org/docs/latest/rules/curly)
