# `hugoalh/prefer-statement-block`

> âœ”ï¸ Default and recommended.

> ğŸ©¹ Automatically fixable.

Prefer curly braces around statement blocks.

## ğŸ”§ Options

*This rule does not have any option.*

## âœï¸ Examples

- ```ts
  /* âŒ INVALID */
  if (foo) foo++;

  /* âœ”ï¸ VALID */
  if (foo) {
    foo++;
  }
  ```
- ```ts
  /* âŒ INVALID */
  if (foo) {
    baz();
  } else qux();

  /* âœ”ï¸ VALID */
  if (foo) {
    baz();
  } else {
    qux();
  }
  ```
- ```ts
  /* âŒ INVALID */
  while (bar)
    baz();

  /* âœ”ï¸ VALID */
  while (bar) {
    baz();
  }
  ```

## ğŸ“š References

- [ESLint rule `curly`](https://eslint.org/docs/latest/rules/curly)
